afonsoft/VisualStudio
# Support PMA (Pixel-based Multi-Anchor) for Visual Studio plugin

Note - 
- Test below scenarios on VS 2019 and VS 2022 
- Test with checked and unchecked of 
   -  Optimize rendering for screens with different pixel densities

**Test  cases** 

- Go to Tools -> Options -> CxViewer -> Authentication -> Login with SAST Server
- Logout SAST Server
- Bind Project
- UnBind Project
- Cancel Bind Project
- Click on Bind Project -> Search Project name 
- Click on Bind Project -> Search Project name  -> Cancel Search project
- Open Documents\Visual Studio (2022\2019)\Settings\CxVSPlugin.conf file -> update BindProjectCount =2 -> again unbind and bind project -> only 2 projects will display to user and user can search project as well
- Run Full Scan Project
- Run Incremental project
- Retrieve Reuslts
- CxViewer scan progress -> click on vulnurablity high/low /medium
- Click on Expand All, Collapse All
- Click on Show description (After clicks opens in IE -> already new ticket added in JIRA)
- Open CxViewer Result -> Change Result state and verify
- Open CxViewer Result -> Change Result severity and verify
- Open CxViewer Result -> Change Assign to user and verify
- Open CxViewer Result -> click on any result it redirects to code
- Open CxViewer Result -> Add comments and verify comment
- Open CxViewer Graph -> click on block it redirects to code(BFL)
- Open CxViewer Path -> click on block it redirects to code(BFL)
- Close VS -> open again -> Project Unbound done
